File: android/src/main/java/com/michaelgerullis/deviceauthenticity/DeviceAuthenticityPlugin.java
package com.michaelgerullis.deviceauthenticity;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import android.content.Context;

@CapacitorPlugin(name = "DeviceAuthenticity")
public class DeviceAuthenticityPlugin extends Plugin {

    private DeviceAuthenticity implementation;

    @Override
    public void load() {
        Context context = getContext();
        implementation = new DeviceAuthenticity(context);
    }

    @PluginMethod
    public void checkAuthenticity(PluginCall call) {
        implementation.checkAuthenticity(call);
    }

    @PluginMethod
    public void isEmulator(PluginCall call) {
        implementation.isEmulator(call);
    }

    @PluginMethod
    public void isRooted(PluginCall call) {
        implementation.isRooted(call);
    }

    @PluginMethod
    public void isInstalledFromAllowedStore(PluginCall call) {
        implementation.checkAuthenticity(call);
    }
    
    @PluginMethod
    public void getApkSignature(PluginCall call) {
        implementation.getApkCertSignature(call);
    }

    @PluginMethod
    public void getApkCertSignature(PluginCall call) {
        implementation.getApkCertSignature(call);
    }

    @PluginMethod
    public void checkApkCertSignature(PluginCall call) {
        implementation.checkApkCertSignature(call);
    }

    @PluginMethod
    public void checkTags(PluginCall call) {
        implementation.checkTags(call);
    }

    @PluginMethod
    public void checkPaths(PluginCall call) {
        implementation.checkPaths(call);
    }

    @PluginMethod
    public void checkExecutableFiles(PluginCall call) {
        implementation.checkExecutableFiles(call);  
    }

    @PluginMethod
    public void isJailbroken(PluginCall call) {
        implementation.isJailbroken(call);
    }

    @PluginMethod
    public void checkPrivateWrite(PluginCall call) {
        implementation.checkPrivateWrite(call);
    }

    @PluginMethod
    public void hasThirdPartyAppStore(PluginCall call) {
        implementation.hasThirdPartyAppStore(call);
    }
}

File: ios/Sources/DeviceAuthenticityPlugin/DeviceAuthenticityPlugin.swift
import Foundation
import Capacitor

@objc(DeviceAuthenticityPlugin)
public class DeviceAuthenticityPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "DeviceAuthenticityPlugin"
    public let jsName = "DeviceAuthenticity"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "checkAuthenticity", returnType: CAPPluginReturnPromise), 
        CAPPluginMethod(name: "isEmulator", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "isJailbroken", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "checkPaths", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "checkPrivateWrite", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "hasThirdPartyAppStore", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "isRooted", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "isInstalledFromAllowedStore", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getApkCertSignature", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "checkApkCertSignature", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "checkTags", returnType: CAPPluginReturnPromise),
    ]
    private let implementation = DeviceAuthenticity()
    @objc func checkAuthenticity(_ call: CAPPluginCall) {
        implementation.checkAuthenticity(call)
    }
    
    @objc func isEmulator(_ call: CAPPluginCall) {
        implementation.isEmulator(call)
    }

    @objc func isJailbroken(_ call: CAPPluginCall) {
        implementation.isJailbroken(call)
    }

    @objc func checkPaths(_ call: CAPPluginCall) {
        implementation.checkPaths(call)
    }
    
    @objc func checkPrivateWrite(_ call: CAPPluginCall) {
        implementation.checkPrivateWrite(call)
    }

    @objc func hasThirdPartyAppStore(_ call: CAPPluginCall) {
        implementation.hasThirdPartyAppStore(call)
    }

    @objc func isRooted(_ call: CAPPluginCall) {
        implementation.isRooted(call)
    }

    @objc func isInstalledFromAllowedStore(_ call: CAPPluginCall) {
        implementation.isInstalledFromAllowedStore(call)
    }

    @objc func getApkCertSignature(_ call: CAPPluginCall) {
        implementation.getApkCertSignature(call)
    }

    @objc func checkApkCertSignature(_ call: CAPPluginCall) {
        implementation.checkApkCertSignature(call)
    }

    @objc func checkTags(_ call: CAPPluginCall) {
        implementation.checkTags(call)
    }
}
    